### PR TITLE
feat: Add `XDG_CONFIG_HOME`/`.config` to search of pixi global manifest path

### PR DESCRIPTION
**Pull request by hoxbro**
Follow up on the question asked here: https://discord.com/channels/1082332781146800168/1307282213770891329

This will search `$XDG_CONFIG_HOME/pixi/pixi-global.toml`, then `~.config/pixi/pixi-global.toml`, and at last, the current location `$PIXI_HOME/manifests/pixi-global.toml`.

The code could be better, so I would love feedback. 

TODO:
- Need to add some tests (not entirely sure how easy this would be with being dependent on the filesystem)
- Update the https://pixi.sh/latest/features/global_tools/#manifest-locations

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
@hoxbro thanks for your contribution!

I've extended your PR and fixed a few problems with it, docs are now present as well.

Also tried to add tests, but had to give up on that. I simply couldn't find a way to set environment variables without influencing other tests.

**Comment by ruben-arts** [comment]
Hey I checked it out, but it will always try to write into the `pixi_home()` dir. As it will always have a Home dir. 

Please check if it exists first.

**Comment by Hofer-Julian** [comment]
> Please check if it exists first.

Should be fixed now

File: docs/features/global_tools.md [hunk 1]
```diff
@@ -88,27 +88,34 @@ exposed = { py3 = "python" } # (2)!
...
-    |--------------|------------------------------------------------------------------------|------------------------------------------------------------------------------------|
-    | 2            | `$PIXI_HOME/manifests/pixi-global.toml`                                | Global manifest in the user home directory. `PIXI_HOME` defaults to `~/.pixi`      |
-    | 1            | `$HOME/.pixi/manifests/pixi-global.toml`                               | User-specific manifest                                                             |
+    | **Priority** | **Location**                                             | **Comments**                                  |
+    |--------------|----------------------------------------------------------|-----------------------------------------------|
+    | 3            | `$PIXI_HOME/manifests/pixi-global.toml`                  | Global manifest in `PIXI_HOME`.               |
+    | 2            | `$HOME/.pixi/manifests/pixi-global.toml`                 | Global manifest in user home directory.       |
+    | 1            | `$HOME/Library/Application Support/pixi/pixi-global.toml`| Config directory.                             |
```
Comment (ruben-arts) [inline docs/features/global_tools.md:108]:
```suggestion
    | 1            | `$HOME/Library/Application Support/pixi/manifests/pixi-global.toml`| Config directory.                             |
```

File: docs/features/global_tools.md [hunk 2]
```diff
@@ -88,27 +88,34 @@ exposed = { py3 = "python" } # (2)!
...
-    | 2            | `$PIXI_HOME\manifests/pixi-global.toml`                                | Global manifest in the user home directory. `PIXI_HOME` defaults to `%USERPROFILE%/.pixi`      |
-    | 1            | `%USERPROFILE%\.pixi\manifests\pixi-global.toml`                       | User-specific manifest                                                                         |
+    | **Priority** | **Location**                                             | **Comments**                                  |
+    |--------------|----------------------------------------------------------|-----------------------------------------------|
+    | 3            | `$PIXI_HOME\manifests/pixi-global.toml`                  | Global manifest in `PIXI_HOME`.               |
+    | 2            | `%USERPROFILE%\.pixi\manifests\pixi-global.toml`         | Global manifest in user home directory.       |
+    | 1            | `%APPDATA%\pixi\pixi-global.toml`                        | Config directory.                             |
```
Comment (ruben-arts) [inline docs/features/global_tools.md:117]:
```suggestion
    | 1            | `%APPDATA%\pixi\manifests\pixi-global.toml`                        | Config directory.                             |
```

**Review by ruben-arts — APPROVED** [review]
Two more small docs fixes and then it looks fine to me!

**Pull request merged**